### PR TITLE
Add Cache

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Changed
 - Guild caching now using Guava's LoadingCache over ExpiringMap

## Removed
 - ExpiringMap

### CODE REVIEW

**Comment by binaryoverload** [comment]
Done @WalshyDev :+1:

**Comment by WalshyDev** [comment]
Have you done any tests of this with our environment? Like 2k+ loaded, see if there's any impact, see if everything works correctly?

File: src/main/java/stream/flarebot/flarebot/FlareBotManager.java [hunk 1]
```diff
@@ -102,28 +93,10 @@ public void executeCreations() {
...
-            public void run() {
-                if (!FlareBot.EXITING.get())
-                    guilds.purge();
-                else
-                    cancel();
-            }
-        }.repeat(30_000, 30_000);
+        guilds = CacheBuilder.newBuilder()
+                .expireAfterAccess(GUILD_EXPIRE, TimeUnit.MILLISECONDS)
+                .removalListener((RemovalListener<String, GuildWrapper>) removalNotification -> saveGuild(removalNotification.getKey(), removalNotification.getValue(), -1))
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/FlareBotManager.java]: why is the lastAccessTime -1 here?

File: src/main/java/stream/flarebot/flarebot/GitHandler.java [hunk 2]
```diff
@@ -20,7 +22,15 @@
     static {
         Git git;
         try {
-            git = Git.open(new File("."));
+            git = Git.open(new File("FlareBot/"));
+        } catch (RepositoryNotFoundException e) {
+            try {
+                git =
+                        Git.cloneRepository().setDirectory(new File("FlareBot/")).setURI("https://github.com/FlareBot/FlareBot.git").call();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/GitHandler.java]: I hate this formatting :D

File: src/main/java/stream/flarebot/flarebot/FlareBotManager.java (hunk 1 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/FlareBotManager.java]: I personally think we should remove lastAccessTime all together. It serves not real use in the new cache system.

**Comment by binaryoverload** [comment]
@WalshyDev If you give me the stuff to load fake data then sure :P

File: src/main/java/stream/flarebot/flarebot/GitHandler.java (hunk 2 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/GitHandler.java]: Ugh me too, I have yet to find a way to disable it

File: src/main/java/stream/flarebot/flarebot/FlareBotManager.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/FlareBotManager.java]: It has nothing to do with the cache system, it is for ordering in Cassandra, means guilds that are active get loaded quickly. It is all for making loading as quick as possible.

File: src/main/java/stream/flarebot/flarebot/GitHandler.java (hunk 2 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/GitHandler.java]: Well mine doesn't do it so IntelliJ formatting settings ;)

**Comment by WalshyDev** [comment]
@binaryoverload Remind me in 3 hours

**Comment by WalshyDev** [comment]
Nice one on reminding me. Here is the way I did it, https://paste.flarebot.stream/nawegiyute.java
Use your imagination to create a couple thousand of fake guilds and things.

**Pull request merged**